Gordon360/Services/DirectMessageService.cs

```diff
@@ -255,6 +256,83 @@ public bool StoreUserRooms(String userId, String roomId)
 
 
 
+            return returnAnswer;
+
+        }
+
+
+        public bool StoreUserConnectionIds(String userId, String connectionId)
+        {
+            var _unitOfWork = new UnitOfWork();
+            var query = _unitOfWork.AccountRepository.FirstOrDefault(x => x.gordon_id == userId);
+            if (query == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found." };
+            }
+
+            var userIdParam = new SqlParameter("@user_id", userId);
+            var connectionIdParam = new SqlParameter("@connection_id", connectionId);
+
+
+            var result = RawSqlQuery<MessageViewModel>.query("INSERT_USER_CONNECTION_ID @user_id, @connection_id", userIdParam, connectionIdParam); //run stored procedure
+
+            bool returnAnswer = true;
+
+            if (result == null)
+            {
+                returnAnswer = false;
+                throw new ResourceNotFoundException() { ExceptionMessage = "The data was not found." };
+            }
+
+
+
+            return returnAnswer;
+
+        }
+
+        public IEnumerable<ConnectionIdViewModel> GetUserConnectionIds(String userId)
+        {
+            var _unitOfWork = new UnitOfWork();
+            var query = _unitOfWork.AccountRepository.FirstOrDefault(x => x.gordon_id == userId);
+            if (query == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The account was not found." };
+            }
+
+            var userIdParam = new SqlParameter("@user_id", userId);
+
+            var result = RawSqlQuery<ConnectionIdViewModel>.query("GET_ALL_CONNECTION_IDS_BY_ID @user_id", userIdParam); //run stored procedure
+
+            var model = result.Select(x =>
+            {
+                ConnectionIdViewModel y = new ConnectionIdViewModel();
+                y.connection_id = x.connection_id;
+               
+
+                return y;
+            });
+
+
+            return model;
+
+        }
+
+        public bool DeleteUserConnectionIds(String connectionId)
+        {
+            var connectionIdParam = new SqlParameter("@connection_id", connectionId);
+
+            var result = RawSqlQuery<MessageViewModel>.query("DELETE_USER_CONNECTION_ID  @connection_id", connectionIdParam); //run stored procedure
+
+            bool returnAnswer = true;
+
+            if (result == null)
+            {
+                returnAnswer = false;
+                throw new ResourceNotFoundException() { ExceptionMessage = "The data was not found." };
+            }
+
+
```

Comment:
Great job on separating each function's purpose. All of them has a specific purpose and are not complex